enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,483 @@
+
+Furthermore, a lot of the rules about the structure of an index are implicitly documented in various sources that are scattered. A best effort is being made to keep the rules updated in the form of a library in the [api repo](https://github.com/operator-framework/api/tree/master/pkg/validation) that is imported by tools like `opm`, `operator-courier` etc that perform index validation. However, like all tools that aspire to provide data validation, these tools, or api libraries that these tools import rules from, have to be constantly maintained to keep up with the latest rule changes, while also maintaining older rules in previous releases. This introduces a lot of overhead in ensuring that olm is easy to use by users, while also leaving windows for lapses due to mis-coordination between different tools across different versions.
+
+Since indexes are manipulated by making additive changes to it, the way an index can be reproduced today is my managing a knowledge base of the series of changes made to the index (i.e storing the `opm` commands chronologically) and by mananging a set of intermediary artifacts (images of the index with their corresponding digest etc). This introduces a lot of overhead in maintaining indexes, and thereby has significantly burdened the `opm` tool with feature requests to add an increasing number of subcommands to allow index authors to manipulate their indexes in various ways. 
+
+### Goals
+
+A human consumable representation of pacakages in an index and the entire index itself in the form of json/yaml will: 
+
+1. Allow for editing of package level information without rebuilding and republishing artifacts(remove bunlde/s from a channel, switch bundle/s to a different channel etc).
+2. Allow algebraic operations to be performed on indexes in a seamless way(combine two indexes, split an index, remove bundles from the index, etc).
+3. Allow for validation of the structure of an index against a standard definition of an index.
+4. Provide a way for index authors to reason about their indexes via a visual representation of the index, without having to curl a grpc api that in turn queries a sqllite database underneath, etc.
+5. Provide a way for tools like `opm`, `operator-registry` etc to consume the representation of an index instead of needing a sqlite database as an input whenever they need information about the index to perform a task. 
+6. Allow for reproducability of an index using just the yaml/json representation of the index. 
+
+### Non-goals 
+
+
+### User Stories 
+
+#### Story 1
+
+As a user with pull permission from the namespace an index is hosted in/as a component of operator-lifecycle-manager, I can query an index for a json/yaml representation of the contents of an index. 
+
+For example, when the community-operators index from the example above is queried for it's representation:
+
+`opm index represent --output=json --index=docker.io/my-namespace/community-operators` 
```

Comment:
Updated with `opm index inspect -o json docker.io/my-namespace/community-operators` 
Since we're operating on an index, seems natural to me that we introduce the command `inspect` or otherwise as a subcommand of `index`. 
@dmesser does this syntax read better for you?